src/initialize.ts

```diff
@@ -48,6 +50,8 @@ const withState = <T, R>(f: (state: State, ...args: T[]) => R) =>
 /** Imports text on given unranked path. Used in puppeteer testing. */
 const importTextFirstMatch = (unrankedPath: string[], text: string) => store.dispatch(importOnFirstMatchPathActionCreator(unrankedPath, text))
 
+store.subscribe(expandOnHoverTop)
```

Comment:
Could you explain a little more the reason for using the subscription api? Why not use a middleware or storeEnhancer? 

Or even simpler, make the appropriate updates in `dragInProgress` (basically the same question as asked above, though I may be missing an obvious reason why that is not viable)?

> If you approve of this pattern I will refactor other hook as well.

If I am understanding you correctly, you suggest the use of the subscription api for both expanding top and bottom, is that right?

Reply:
> Could you explain a little more the reason for using the subscription api? Why not use a middleware or storeEnhancer?

You are correct. It can be easily converted to an action creator.